docs/hosting/deploy.md

```diff
@@ -1,20 +1,417 @@
 
-Please follow the hosting guide here to deploy your app **(invitation code needed)**: [Reflex Hosting](https://reflex-dev.notion.site/Reflex-Hosting-Documentation-57a4dd55d6234858bbae0be75be79ce7?pvs=4)
+### Authentication Commands
 
+#### reflex login
 
+When you type the `reflex login` command for the very first time, it opens the hosting service login page in your browser. We authenticate users through OAuth. At the moment the supported OAuth providers are Github and Gmail. You should be able to revoke such authorization on your Github and Google account settings page. We do not log into your Github or Gmail account. OAuth authorization provides us your email address and in case of Github your username handle. We use those to create an account for you. The email used in the original account creation is used to identify you as a user. If you have authenticated using different emails, those create separate accounts. To switch to another account, first log out using the `reflex logout` command. More details on the logout command are in [reflex logout](#reflex-logout) section.
+
+```python eval
+doccmdoutput(
+    command="reflex login",
+    output="""Opening https://control-plane.reflex.run ...
+Successfully logged in.
+""",
+)
+```
+
+After authentication, the browser redirects to the original hosting service login page. It shows that you have logged in. Now you can return to the terminal where you type the login command. It should print a message such as `Successfully logged in`.
+
+Your access token is cached locally in the reflex support directory. For subsequent login commands, the cached token is validated first. If the token is still valid, the CLI command simply shows `You’re already logged in`. If the token is expired or simply not valid for any reason, the login command tries to open your browser again for web based authentication.
+
+#### reflex logout
+
+When you successfully authenticate with the hosting service, there is information cached in two different places: a file containing the access token in the reflex support directory, and cookies in your browser. The cookies include the access token, a refresh token, some unix epochs indicating when the access token expires. The logout command removes the cached information from these places.
+
+### Deployment Commands
+
+#### reflex deploy
+
+This is the command to deploy a reflex app from its top level app directory. This directory contains a `rxconfig.py` where you run `reflex init` and `reflex run`. A `requirements.txt` file is required. The deploy command checks the existence of this file and also the content of it (this is available in more recent versions of `reflex-hosting-cli` such as `0.1.3`). The command prompts you for potential updates on the requirements when there are new packages or newer versions of packages detected in your Python environment.
```

Comment:
need more commentary on how the automatic updating of the requirements.txt file works